vidarr-pluginapi/src/main/java/ca/on/oicr/gsi/vidarr/InputType.java

```diff
@@ -754,16 +803,15 @@ public static InputType taggedUnion(Pair<String, InputType>... elements) {
    * @param elements the possible data structures; the string identifiers must be unique
    */
   public static InputType taggedUnionFromPairs(Stream<Pair<String, InputType>> elements) {
-    return new InputType() {
-      private final Map<String, InputType> union =
-          Collections.unmodifiableMap(
-              elements.collect(Collectors.toMap(Pair::first, Pair::second)));
-
-      @Override
-      public <R> R apply(Visitor<R> transformer) {
-        return transformer.taggedUnion(union.entrySet().stream());
-      }
-    };
+    return new TaggedUnionInputType(
+        elements.collect(
+            Collectors.toMap(
+                (e) -> ((String) ((Pair) e).first()),
```

Comment:
Why are these casts here? `Pair::first` and `Pair::second` should get you the right type.

Reply:
Pair::first etc complained about non-static methods being called statically or something, and then .first() returned only Object which messed things up further down. I will see if this resolves with the other fixes you've suggested

Reply:
That usually happens when there's another type error and the compiler gets stuck in type-inference land.

Reply:
Idk why it didn't work before, but it does now!